lib/Runtime/Library/amd64/JavascriptFunctionA.S

```diff
@@ -2,27 +2,41 @@
 // Copyright (C) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
+
 .intel_syntax noprefix
+#include "unixasmmacros.inc"
 
-#include "pal/inc/unixasmmacros.inc"
-        
 .extern _ZN2Js18JavascriptFunction13DeferredParseEPPNS_14ScriptFunctionE
 
-.global C_FUNC(amd64_CallFunction)
 .global	_ZN2Js18JavascriptFunction20DeferredParsingThunkEPNS_16RecyclableObjectENS_8CallInfoEz
 
 #ifndef __APPLE__
-.type C_FUNC(amd64_CallFunction), @function
 .type _ZN2Js18JavascriptFunction20DeferredParsingThunkEPNS_16RecyclableObjectENS_8CallInfoEz, @function
 #endif
-        
-.text
-C_FUNC(amd64_CallFunction):
+
+//------------------------------------------------------------------------------
+// Invoke
+//
+//      JavascriptMethod(RecyclableObject* function, CallInfo callInfo, ...)
+//
+// with a custom calling convention in order to support JavascriptStackWalker
+// and RUNTIME_ARGUMENTS. JavascriptMethod entry stack layout:
+//
+//      [Return Address]
+//      [function]              == RDI
```

Comment:
Arent the callee save registers in between the Return address and function object?